src/assets/js/components/session.tsx

```diff
@@ -83,12 +90,26 @@ export default class SessionComponent extends React.Component<Props, State> {
     // make true to output time taken to get render contents
     this.profileRender = true;
 
+    this.getProfiler = (profileRender) => {
+      if (!profileRender) {
+        return {
+          start: () => null,
+          finish: () => null
+        };
+      }
+
+      let t0 = Date.now();
+
+      return {
+        start: () => { t0 = Date.now(); },
+        finish: () => logger.info('Update took time', Date.now() - t0)
```

Comment:
in theory, this profiler can't be used twice at the same time.  this isn't an issue since the updates are debounced, but it'd be good to not make that assumption.  perhaps start could return finish.  Also it should be possible to put a message saying what's being profiled - note the "forceLoadTree" got dropped